docs/app/utils/server/doc.server.ts

```diff
@@ -52,41 +37,24 @@ const stripTrailingSlashes = (str: string): string => {
 }
 
 export const getParsedMetadata = async (tag: string) => {
-  if (process.env.NODE_ENV === 'development') {
-    const content = await readFile(
-      resolve(_dirname, '../../../', `posts/${tag}/metadata.json`),
-      'utf-8'
-    )
-
-    if (!content) {
-      return null
-    }
+  const content = await readFile(
+    resolve(appRootPath, `posts/${tag}/metadata.json`),
+    'utf-8'
+  )
 
-    return MetadataSchema.parse(JSON.parse(content))
+  if (!content) {
+    return null
   }
 
-  const promise = await s3.getObject({
-    Bucket: process.env.AWS_S3_BUCKET || '',
-    Key: `posts/${tag}/metadata.json`,
-  })
-
-  const content =
-    (await promise.Body?.transformToString()) ||
-    '{\npaths: {},\nhasIndex: false,\nsections: [],\nmeta: {},\n}'
-
   return MetadataSchema.parse(JSON.parse(content))
 }
 
 export const tagHasIndex = async (tag: string) => {
-  if (process.env.NODE_ENV === 'development') {
-    // use metadata to handle the cross-checking
-    // const metadata = await getParsedMetadata(tag)
-    // return metadata?.hasIndex
-    // or
-    return existsSync(resolve(_dirname, '../../../', `posts/${tag}/_index.mdx`))
-  }
-
-  return ((await getParsedMetadata(tag)) ?? {}).hasIndex
```

Comment:
Did i delete a line too much?

Reply:
probably not, seeing that the development one returned early